app/assets/javascripts/models/data/project.js.coffee

```diff
@@ -87,9 +87,12 @@ ns.Project = do (ko,
       options = @serviceOptions.serialize()
       $(document).trigger('dataaccessevent', [@collection.id, options])
 
+      form_data = []
+      form_data.push method.form for method in @granuleAccessOptions().methods
       id: @collection.id
       params: param(@collection.granuleDatasource()?.toQueryParams() ? @collection.query.globalParams())
       serviceOptions: options
+      form: form_data
```

Comment:
I agree that sending hash to rails backend is the cleaner way! On the other hand, sha-1 will be introduced in addition to murmur3. I used murmur3 to hash the excluded cwic granule ids since it is shorter than sha-1 and the collision rate is acceptable. I used sha-1 in rails because it is just an easy grab from the digest lib. Should I switch to use murmur3 in rails as well?

I like the idea of hashing the whole form more than just the selected (easier :))..

Is there a way to guarantee that the form is not dirty?